core/pow/engine.go

```diff
@@ -561,3 +561,25 @@ func (e *Engine) BlockData() (uint64, string) {
 	}
 	return e.currentBlock, e.blockHash[e.currentBlock%ringSize]
 }
+
+func (e *Engine) GetSpamStatistics(partyID string) (uint64, int64) {
+	e.lock.RLock()
+	defer e.lock.RUnlock()
+
+	totalTx := uint64(0)
+
+	for _, state := range e.activeStates {
```

Comment:
I don't think this is sufficient. I think for pow what I'd want to know as a user is for every block height that I have transactions for, how many transactions I've sent and what is the total difficulty of them. So the statistics looks like, a list of blocks, for each block, number of transactions and total difficulty. you can also throw in the required difficulty for this block height. anything else? @davidsiska-vega @TomMcL

Reply:
:+1: . Not even sure whether you need total difficulty given you can check you aren't currently blocked, but it can't hurt and might make some wallet-side calculations easier

Reply:
Addressed in https://github.com/vegaprotocol/vega/pull/7275/commits/9804a5d4f1f2567c57a6ee0713cb1ea7da56d315